diff --git a/lua/cmp_buffer/buffer.lua b/lua/cmp_buffer/buffer.lua
index d0ed285..4a02745 100644
--- a/lua/cmp_buffer/buffer.lua
+++ b/lua/cmp_buffer/buffer.lua
@@ -119,7 +119,7 @@ function buffer.index_range(self, range_start, range_end, skip_already_indexed)
     local chunk_start = range_start
     while chunk_start < range_end do
       local chunk_end = math.min(chunk_start + chunk_size, range_end)
-      local chunk_lines = vim.api.nvim_buf_get_lines(self.bufnr, chunk_start, chunk_end, true)
+      local chunk_lines = vim.api.nvim_buf_get_lines(self.bufnr, chunk_start, chunk_end, false)
       for i, line in ipairs(chunk_lines) do
         if not skip_already_indexed or not self.lines_words[chunk_start + i] then
           self:index_line(chunk_start + i, line)
