diff --git a/lua/copilot/suggestion.lua b/lua/copilot/suggestion.lua
index cad0c0b..3d03e4f 100644
--- a/lua/copilot/suggestion.lua
+++ b/lua/copilot/suggestion.lua
@@ -568,11 +568,21 @@ local function on_buf_enter()
   end
 end
 
+local last_cursor_moved
+
 local function on_cursor_moved_i()
-  local ctx = get_ctx()
-  if copilot._copilot_timer or ctx.params or should_auto_trigger() then
-    schedule(ctx)
-  end
+  clear()
+  local cursor_moved = vim.uv.now()
+  last_cursor_moved = cursor_moved
+  vim.defer_fn(function()
+    if last_cursor_moved ~= cursor_moved then
+      return
+    end
+    local ctx = get_ctx()
+    if copilot._copilot_timer or ctx.params or should_auto_trigger() then
+      schedule(ctx)
+    end
+  end, 16)
 end
 
 local function on_text_changed_p()
