diff --git a/lua/fzf-lua/utils.lua b/lua/fzf-lua/utils.lua
index 0a56b4e..5020b77 100644
--- a/lua/fzf-lua/utils.lua
+++ b/lua/fzf-lua/utils.lua
@@ -429,6 +429,10 @@ end
 function M.hexcol_from_hl(hlgroup, what)
   if not hlgroup or not what then return end
   local hexcol = synIDattr(hlgroup, what)
+  -- some colorschemes set fg=fg/bg or bg=fg/bg
+  if hexcol == "fg" or hexcol == "bg" then
+    return ""
+  end
   if hexcol and not hexcol:match("^#") then
     -- try to acquire the color from the map
     -- some schemes don't capitalize first letter?
