diff --git a/lua/gitsigns/async.lua b/lua/gitsigns/async.lua
index 17cbff9..d278fe5 100644
--- a/lua/gitsigns/async.lua
+++ b/lua/gitsigns/async.lua
@@ -73,9 +73,7 @@ local function run(func, callback, ...)
 
     if not stat then
       local err = ret[2] --[[@as string]]
-      error(
-        string.format('The coroutine failed with this message: %s\n%s', err, debug.traceback(co))
-      )
+      return
     end
 
     if coroutine.status(co) == 'dead' then
