diff --git a/lua/lazy/core/config.lua b/lua/lazy/core/config.lua
index 9319999..0a7da68 100644
--- a/lua/lazy/core/config.lua
+++ b/lua/lazy/core/config.lua
@@ -83,6 +83,7 @@ M.defaults = {
     icons = {
       cmd = " ",
       config = "",
+      debug = "●",
       event = " ",
       favorite = " ",
       ft = " ",
diff --git a/lua/lazy/manage/task/git.lua b/lua/lazy/manage/task/git.lua
index 4bd7134..d3e2629 100644
--- a/lua/lazy/manage/task/git.lua
+++ b/lua/lazy/manage/task/git.lua
@@ -198,13 +198,18 @@ M.status = {
             return line ~= ""
           end, lines)
           if #lines > 0 then
-            local msg = { "You have local changes in `" .. self.plugin.dir .. "`:" }
-            for _, line in ipairs(lines) do
-              msg[#msg + 1] = "  * " .. line
+            local patch_path = vim.fs.joinpath(vim.fn.stdpath("config"), "patches", self.plugin.name .. ".patch")
+            local patch_stat = vim.uv.fs_stat(patch_path)
+            -- Do not warn about local changes if there is a patch file
+            if not patch_stat or patch_stat.type ~= "file" then
+              local msg = { "You have local changes in `" .. self.plugin.dir .. "`:" }
+              for _, line in ipairs(lines) do
+                msg[#msg + 1] = "  * " .. line
+              end
+              msg[#msg + 1] = "Please remove them to update."
+              msg[#msg + 1] = "You can also press `x` to remove the plugin and then `I` to install it again."
+              self:error(msg)
             end
-            msg[#msg + 1] = "Please remove them to update."
-            msg[#msg + 1] = "You can also press `x` to remove the plugin and then `I` to install it again."
-            self:error(msg)
           end
         end
       end,
diff --git a/lua/lazy/view/render.lua b/lua/lazy/view/render.lua
index e1eec6c..b42d3ea 100644
--- a/lua/lazy/view/render.lua
+++ b/lua/lazy/view/render.lua
@@ -761,7 +761,7 @@ function M:debug()
         ---@type string[]
         plugins = vim.tbl_values(plugins)
         table.sort(plugins)
-        self:append("● ", "LazySpecial", { indent = 2 })
+        self:append(Config.options.ui.icons.debug, "LazySpecial", { indent = 2 })
         if handler_type == "keys" then
           for k, v in pairs(Config.plugins[plugins[1]]._.handlers.keys) do
             if k == value then
