diff --git a/lua/lazy/core/config.lua b/lua/lazy/core/config.lua
index 1fbf3ef..0522226 100644
--- a/lua/lazy/core/config.lua
+++ b/lua/lazy/core/config.lua
@@ -55,6 +55,7 @@ M.defaults = {
     icons = {
       cmd = " ",
       config = "",
+      debug = "●",
       event = " ",
       ft = " ",
       init = " ",
diff --git a/lua/lazy/view/render.lua b/lua/lazy/view/render.lua
index 7487056..029be63 100644
--- a/lua/lazy/view/render.lua
+++ b/lua/lazy/view/render.lua
@@ -686,7 +686,7 @@ function M:debug()
         ---@type string[]
         plugins = vim.tbl_values(plugins)
         table.sort(plugins)
-        self:append("● ", "LazySpecial", { indent = 2 })
+        self:append(Config.options.ui.icons.debug, "LazySpecial", { indent = 2 })
         if handler_type == "keys" then
           for k, v in pairs(Config.plugins[plugins[1]]._.handlers.keys) do
             if k == value then
