diff --git a/lua/lazy/core/config.lua b/lua/lazy/core/config.lua
index 5aa61cb..b1187e6 100644
--- a/lua/lazy/core/config.lua
+++ b/lua/lazy/core/config.lua
@@ -58,6 +58,7 @@ M.defaults = {
     icons = {
       cmd = " ",
       config = "",
+      debug = "●",
       event = " ",
       favorite = " ",
       ft = " ",
diff --git a/lua/lazy/manage/task/git.lua b/lua/lazy/manage/task/git.lua
index 8dd2536..c6a7650 100644
--- a/lua/lazy/manage/task/git.lua
+++ b/lua/lazy/manage/task/git.lua
@@ -190,12 +190,17 @@ M.status = {
             return line ~= ""
           end, lines)
           if #lines > 0 then
-            self.error = "You have local changes in `" .. self.plugin.dir .. "`:\n"
-            for _, line in ipairs(lines) do
-              self.error = self.error .. "  * " .. line .. "\n"
+            local patch_path = vim.fs.joinpath(vim.fn.stdpath("config"), "patches", self.plugin.name .. ".patch")
+            local patch_stat = vim.uv.fs_stat(patch_path)
+            -- Do not warn about local changes if there is a patch file
+            if not patch_stat or patch_stat.type ~= "file" then
+              self.error = "You have local changes in `" .. self.plugin.dir .. "`:\n"
+              for _, line in ipairs(lines) do
+                self.error = self.error .. "  * " .. line .. "\n"
+              end
+              self.error = self.error .. "Please remove them to update.\n"
+              self.error = self.error .. "You can also press `x` to remove the plugin and then `I` to install it again."
             end
-            self.error = self.error .. "Please remove them to update.\n"
-            self.error = self.error .. "You can also press `x` to remove the plugin and then `I` to install it again."
           end
         end
       end,
diff --git a/lua/lazy/view/render.lua b/lua/lazy/view/render.lua
index 5d7e197..2a021eb 100644
--- a/lua/lazy/view/render.lua
+++ b/lua/lazy/view/render.lua
@@ -686,7 +686,7 @@ function M:debug()
         ---@type string[]
         plugins = vim.tbl_values(plugins)
         table.sort(plugins)
-        self:append("● ", "LazySpecial", { indent = 2 })
+        self:append(Config.options.ui.icons.debug, "LazySpecial", { indent = 2 })
         if handler_type == "keys" then
           for k, v in pairs(Config.plugins[plugins[1]]._.handlers.keys) do
             if k == value then
