diff --git a/plugin/lspconfig.lua b/plugin/lspconfig.lua
index 1a731eb..9db9f71 100644
--- a/plugin/lspconfig.lua
+++ b/plugin/lspconfig.lua
@@ -86,10 +86,25 @@ end, {
   complete = lsp_complete_configured_servers,
 })
 
+local function stop_client(client, force, num_trials)
+  num_trials = num_trials or 4
+  if client.is_stopped() then
+    return
+  end
+  if force or num_trials <= 0 then
+    client.stop(true)
+    return
+  end
+  client.stop()
+  vim.defer_fn(function()
+    stop_client(client, force, num_trials - 1)
+  end, 500)
+end
+
 api.nvim_create_user_command('LspRestart', function(info)
   local detach_clients = {}
   for _, client in ipairs(get_clients_from_cmd_args(info.args)) do
-    client.stop()
+    stop_client(client)
     if vim.tbl_count(client.attached_buffers) > 0 then
       detach_clients[client.name] = { client, client.attached_buffers }
     end
@@ -136,12 +151,12 @@ api.nvim_create_user_command('LspStop', function(info)
     local servers_on_buffer = lsp.get_active_clients { bufnr = current_buf }
     for _, client in ipairs(servers_on_buffer) do
       if client.attached_buffers[current_buf] then
-        client.stop(force)
+        stop_client(client, force)
       end
     end
   else
     for _, client in ipairs(get_clients_from_cmd_args(server_id)) do
-      client.stop(force)
+      stop_client(client, force)
     end
   end
 end, {
