diff --git a/lua/oil/keymap_util.lua b/lua/oil/keymap_util.lua
index 5f8f321..172dc9d 100644
--- a/lua/oil/keymap_util.lua
+++ b/lua/oil/keymap_util.lua
@@ -111,7 +111,7 @@ M.show_help = function(keymaps)
     height = math.min(editor_height, #lines),
     zindex = 150,
     style = "minimal",
-    border = "rounded",
+    border = vim.g.modern_ui and "solid" or "single",
   })
   local function close()
     if vim.api.nvim_win_is_valid(winid) then
