diff --git a/autoload/rnvimr.vim b/autoload/rnvimr.vim
index e9ffb8f..452cc14 100644
--- a/autoload/rnvimr.vim
+++ b/autoload/rnvimr.vim
@@ -158,8 +158,17 @@ function! rnvimr#toggle() abort
                 call nvim_win_close(win_hd, 0)
                 call rnvimr#rpc#clear_image()
             else
-                call nvim_set_current_win(win_hd)
-                startinsert
+                for win in nvim_tabpage_list_wins(0)
+                    if win == win_hd
+                        call nvim_set_current_win(win_hd)
+                        startinsert
+                        return
+                    endif
+                endfor
+                call nvim_win_close(win_hd, 0)
+                call rnvimr#rpc#clear_image()
+                call rnvimr#rpc#attach_file_once(expand('%:p'))
+                call s:reopen_win()
             endif
         else
             call rnvimr#rpc#attach_file_once(expand('%:p'))
